### PR TITLE
FrontEnd: Display of All Products Component Using Product Component and Sort Component 

### LINKED ISSUE #44: FrontEnd: Display of All Products Component Using Product Component and Sort Component
## **User story**
Who: As a developer
What: I want to be have a Products Display component with Sorting option 
Why: so that user can view all products and sort them how user wants to see

## Acceptance criteria
Create a component for display products and able to sort them like the one we can see below
Screenshot 2022-01-29 at 08 32 06

Check the wireframes here:
https://www.figma.com/file/yo4QT2EcPo83HAMI3ydMxb/HYF-Wireframes?node-id=216%3A28468

## Implementation details
Display products with sorting option using Products Display Component and Sort Component

### PR DESCRIPTION
**Pull request by suvarna-ratna**
# Description
This PR displays Products with Sorting option, so that user can view all products and sort them how user wants to see

Please provide a short summary explaining what this PR is about.

Fixes #44 

# How to test?
npm run storybook
Please provide a short summary how your changes can be tested?

# Checklist

- [X] I have performed a self-review of my own code
- [X] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [X] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [X] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [X] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/ProductsComponent/ProductsComponent.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  };
+  return (
+    <div className="products-component">
+      <SortComponent
+        products={products}
+        setSortedProducts={setSortedProducts}
+      />
+
+      <div>
+        <ul className="product-list">
```
Comment (senner007) [inline src/client/components/ProductsComponent/ProductsComponent.js:23]: I would create one more abstraction here in the form of a Products component here.

File: src/client/components/ProductsComponent/ProductsComponent.stories.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+export default {
+  title: 'Components / Products Component',
+  component: ProductsComponent,
+  argTypes: {
+    products: { control: '' },
+  },
+};
+
+const template = ({ products }) => {
+  return (
```
Comment (senner007) [inline src/client/components/ProductsComponent/ProductsComponent.stories.js:14]: Is there a reason why you choose not to showcase the sorting here also?

File: README.md [hunk 3]
```diff
@@ -71,7 +70,7 @@ This repository is Class19's Final Project which is Plant Webshop
 
 ## 👩🏻‍💼💡👨🏽‍💼 The Customer
```
Comment (senner007) [inline README.md]: restore the changes to this file

**Review by senner007 — COMMENTED** [review]
I havent run the code. But it seems good 👍

**Review by santhoshboinapally — COMMENTED** [review]
image (1)
I run the code its working fine on machine, Missing images. Good Job.

File: README.md (hunk 3 above)
Comment (suvarna-ratna) [reply README.md]: done .. Tak

File: src/client/components/ProductsComponent/ProductsComponent.stories.js (hunk 2 above)
Comment (suvarna-ratna) [reply src/client/components/ProductsComponent/ProductsComponent.stories.js:14]: Got error i think .. with update of sorted products.. 
Thats why I did like this..

File: src/client/components/ProductsComponent/ProductsComponent.js (hunk 1 above)
Comment (suvarna-ratna) [reply src/client/components/ProductsComponent/ProductsComponent.js:23]: Okay ..

File: src/client/components/ProductsComponent/ProductsComponent.stories.js (hunk 2 above)
Comment (senner007) [reply src/client/components/ProductsComponent/ProductsComponent.stories.js:14]: Sorry, my mistake. The sorting is working here.

**Review by senner007 — APPROVED** [review]
👌Great work.

If we want to get nit-picky with everything - and we do as we progress - then we might improve the sort component also. It both displays the sorting menu and does the sorting logic. A better approach would be to separate the concerns and maybe create a useSorting custom hook for the sorting logic, even though its not reused anywhere.

The _single responsibility principle_ tells us that everything must be separated as much as possible. The more the better. Thinking like this is the secret to correct programming, imo.

**Review by Divyajg — COMMENTED** [review]
Works fine on my machine.

**Review by Divyajg — APPROVED** [review]

**Review by JiuTak — CHANGES_REQUESTED** [review]
It is working well in my machine :)

I am wondering sorting shouldn't be the other way?
sort = price down 
then shouldn't it be a high price to lower?

image

**Comment by suvarna-ratna** [comment]
@JiuTak , high to low is also there.... plz check

**Review by santhoshboinapally — APPROVED** [review]

**Comment by JiuTak** [comment]
> @suvarna-ratna  , high to low is also there.... plz check

I know,,, it sorting the other way I mean......
price ↓ = high to low
price ↑ = should be low to high 
am I understand wrong?

**Comment by suvarna-ratna** [comment]
> > @suvarna-ratna  , high to low is also there.... plz check
> 
> I know,,, it sorting the other way I mean...... price ↓ = high to low price ↑ = should be low to high am I understand wrong?

Thank you .. I changed arrows

**Pull request merged**